File: CATCHOPP/MS_CompetenceAndReview/src/main/java/org/example/ms_competenceandreview/Services/Interface/CompetanceService.java
package org.example.ms_competenceandreview.Services.Interface;

import org.example.ms_competenceandreview.Entities.Competance;

import java.util.List;

public interface CompetanceService {
    Competance AjouterCompetance(Competance competance);
    Competance ModifierCompetance(Competance competance);
    void SupprimerCompetance(Long id);
    Competance GetCompetance(Long id);
    List<Competance> GetAllCompetance();
}

File: FrontFreelanceApp/src/app/Services/competance.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Competance } from '../models/Competance';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CompetanceService {

   constructor(private http:HttpClient) { }  
    private url ='http://localhost:8084/Competance'

parseCv(file: File): Observable<Competance[]> {
  const formData: FormData = new FormData();
  formData.append('file', file);
  
  // On envoie le fichier au nouveau endpoint qu'on a créé
  return this.http.post<Competance[]>(`${this.url}/ParseCV`, formData);
}

    getAllCompetances(): Observable<Competance[]> {
    return this.http.get<Competance[]>(`${this.url}/GetAllCompetance`);
  }

  addCompetance(competance: Competance): Observable<Competance> {
    return this.http.post<Competance>(`${this.url}/AjouterCompetance`, competance);
  }

  updateCompetance(competance: Competance): Observable<Competance> {
    return this.http.put<Competance>(`${this.url}/ModifierCompetance`, competance);
  }

  deleteCompetance(id: number): Observable<void> {
    return this.http.delete<void>(`${this.url}/SupprimerCompetance/${id}`);
  }
}

